app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -30,8 +30,17 @@ def verify_credentials(_auth_type = nil, options = {})
         # As the connect will only give a handle
         # we verify the credentials via an actual operation
         connect(options).inventory.list_feeds
+      rescue URI::InvalidComponentError
+        raise MiqException::MiqHostError, "Host '#{hostname}' is invalid"
+      rescue ::Hawkular::BaseClient::HawkularConnectionException
+        raise MiqException::MiqUnreachableError, "Unable to connect to #{hostname}:#{port}"
+      rescue ::Hawkular::BaseClient::HawkularException => he
+        raise MiqException::MiqInvalidCredentialsError, 'Invalid credentials' if he.status_code == 401
+        raise MiqException::MiqInvalidCredentialsError, 'Hawkular not found on host' if he.status_code == 404
+        raise MiqException::MiqCommunicationsError, he.message
       rescue => err
-        raise MiqException::MiqInvalidCredentialsError, 'Invalid credentials'
+        $log.log_backtrace(err)
```

Comment:
If i understand it correctly, the root cause of the other exceptions aren't logged. Do you think those should be logged?